File: Happy Number/Happy number.cc
class Solution {
public:
    bool isHappy(int n) {
        int sum=0;
        int remainder;
        while (n<10){
            if(n==1||n==7){
                return 1;
                }
            else{
                return 0;
                }
        }
        while (n>0){
            remainder=n%10;
            sum+=pow(remainder, 2);
            n/=10;
        }
        return isHappy(sum);
    }
};

File: Happy Number/Happy number.py
class Solution(object):
    def isHappy(self, n):
        """
        :type n: int
        :rtype: bool
        """
        if n<10:
            if n==1 or n==7:
                return True
            else:
                return False
        total = 0
        while n>10:
            remainder = n%10
            total += pow(remainder, 2)
            n/=10
        return isHappy(total)
            

